pkg/migration/controllers/migration.go

```diff
@@ -1547,18 +1567,55 @@ func (m *MigrationController) prepareCRDClusterResource(
 					return fmt.Errorf("unable to find suspend path, err: %v", err)
 				}
 				disableVersion = int64(0)
+				if curr == 0 && suspendAnnotationIsPresent {
+					//suspendAnnotation has value set as {currVal + "," + suspend.Value}
+					//we need to extract only the currVal from it
+					annotationValue := strings.Split(SuspendAnnotationValue, ",")[0]
+					intValue, err := strconv.ParseInt(annotationValue, 10, 64)
+					if err != nil {
+						return err
+					}
+					if intValue > 0 {
+						//Only if the actual suspend path value is 0 and suspend annotation value is more than 0,
+						//then carry forward the annotation.
+						curr = intValue
+						//we reset the actual suspend path value as well in case the migration has startApplications enabled
+						err = unstructured.SetNestedField(content, fmt.Sprintf("%v", curr), fields...)
+						if err != nil {
+							return err
+						}
+					}
+				}
 				currVal = fmt.Sprintf("%v", curr)
 			} else if suspend.Type == "string" {
 				curr, _, err := unstructured.NestedString(content, fields...)
 				if err != nil {
 					return fmt.Errorf("unable to find suspend path, err: %v", err)
 				}
 				disableVersion = suspend.Value
+				if curr == suspend.Value && suspendAnnotationIsPresent {
+					annotationValue := strings.Split(SuspendAnnotationValue, ",")[0]
+					if annotationValue != "" {
+						//Only if the actual value is equal to suspend.Value and suspend path annotation value is not an empty string,
+						//then carry forward the annotation.
+						curr = annotationValue
+						err = unstructured.SetNestedField(content, curr, fields...)
+						if err != nil {
+							return err
+						}
+					}
+				}
 				currVal = curr
 			} else {
 				return fmt.Errorf("invalid type %v to suspend cr", suspend.Type)
 			}
 
+			//No need to scale down if StartApplications is set to True in the migration spec
+			if *migration.Spec.StartApplications {
```

Comment:
Why is this required, can you please explain?

Reply:
In the existing implementation this check is added at the beginning of the functions (#1519 in previous code) since we do not need to scale down the resource if we are going start the applications in the target cluster.

Since as part of this bug, we may want a different value of replica compared to the existing count (which could be 0 and the real useful value is stored in the annotation), I have moved the check to a different place.

I have explained how we get the appropriate value for this in the above comment with an example.